src/Chapter3.hs

```diff
@@ -753,6 +864,21 @@ parametrise data types in places where values can be of any general type.
   maybe-treasure ;)
 -}
 
+data TreasureChest a = TreasureChest
+    { treasureChestGold :: Int
+    , treasureChestLoot :: a
+    }
+
+data Treasure a = Treasure
+    { treasureHoard :: a
+    }
+
+data Dragon a = Dragon
+    { dragonPower :: a
+    }
+
+data Lair p t = Lair (Dragon p) (Maybe (TreasureChest t)) (Maybe (Treasure t))
```

Comment:
Great!

You could also give names to these fields as these are going to be useful while working with such non-trivial types 👍🏼